apps/sms/test/marionette/lib/views/inbox/view.js

```diff
@@ -11,28 +11,56 @@ function InboxView(client) {
 
 InboxView.prototype = {
   get conversations() {
-    return this.accessors.conversations.map(function(conversation) {
-      return {
-        lastMessageType: conversation.getAttribute('data-last-message-type'),
-        title: this.accessors.getConversationTitle(conversation).text()
-      };
-    }, this);
+    return this.accessors.conversations.reduce(function(list, conversation) {
+      // It's workaround for the case when conversation node reference is stale,
+      // so when we try to access MarionetteElement from the node that's not
+      // attached to the dom we'll get an exception.
+      try {
+        var bodyText = conversation.findElement('.body-text');
+        list.push({
+          id: conversation.getAttribute('data-thread-id'),
+          lastMessageType: conversation.getAttribute('data-last-message-type'),
+          title: this.accessors.getConversationTitle(conversation).text(),
+          bodyText: bodyText.displayed() ? bodyText.text() : '',
+          isDraft: !!conversation.getAttribute('data-draft-id'),
+          hasDraft: conversation.getAttribute('class').indexOf('draft') !== -1
+        });
+      } catch(e) {
+        console.error('Conversation node is not available', e);
+      }
+
+      return list;
+    }.bind(this), []);
   },
 
-  goToFirstThread: function() {
-    this.accessors.firstConversation.tap();
-    var ConversationView = require('../conversation/view');
-    var conversation = new ConversationView(this.client);
-    conversation.accessors.waitToAppear();
-    return conversation;
+  goToConversation: function(conversationId) {
+    var conversation = this.accessors.findConversation(conversationId);
+
+    conversation.tap();
+
+    return !!conversation.getAttribute('data-draft-id') ?
```

Comment:
Mmm honestly it sounds too complex for the integration test purposes..., especially that we need to wait for *that unknown view*  to appear before accessing any of its attributes.

But if you want I can add something like ```parse``` method [1], that will be used in ```conversations``` getter and here. Then here I can parse conversation node to object with handy boolean field ```isDraft```. Essentially it will be the same, but a bit more readable.

What do you prefer?

[1] https://github.com/mozilla-b2g/gaia/blob/e42f7e9937b62f84d2baef547f5bd08e1a6ac44b/apps/sms/test/marionette/lib/views/conversation/message_accessors.js#L23